smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/AbstractMediator.java

```diff
@@ -65,13 +117,48 @@ public void initialize(Object bean) {
                     try {
                         return this.configuration.getMethod().invoke(bean, args);
                     } catch (Exception e) {
+                        e.printStackTrace();
                         throw ex.processingException(configuration.methodAsString(), e);
                     }
                 };
             }
         }
     }
 
+    @SuppressWarnings("unchecked")
+    protected <T> T invoke(Message<?> message) {
+        try {
+            Objects.requireNonNull(this.invoker, msg.invokerNotInitialized());
+            Object[] objects = invocationParameters.apply(message);
+            return (T) this.invoker.invoke(objects);
+        } catch (RuntimeException e) { // NOSONAR
+            log.methodException(configuration().methodAsString(), e);
+            throw e;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected <T> Uni<T> invokeBlocking(Message<?> message) {
+        try {
+            Objects.requireNonNull(this.invoker, msg.invokerNotInitialized());
```

Comment:
It would be nice if there was a way to not have to duplicate `invokeBlocking`, but I see you're wanting to do the parameter work in the blocking thread too.

Reply:
A bit tricky, but yes the duplication is not great.